src/rt/trace.d

```diff
@@ -380,68 +363,160 @@ static void trace_init()
 /////////////////////////////////
 // Terminate.
 
-void trace_term()
+static ~this()
 {
-    //printf("trace_term()\n");
-    if (trace_inited == 1)
-    {   Stack *n;
+    // Free remainder of the thread local stack
+    while (trace_tos)
+    {
+        auto n = trace_tos.prev;
+        stack_free(trace_tos);
+        trace_tos = n;
+    }
 
-        trace_inited = 2;
+    // And free the thread local stack's memory
+    while (stack_freelist)
+    {
+        auto n = stack_freelist.prev;
+        stack_free(stack_freelist);
+        stack_freelist = n;
+    }
+
+    synchronized
```

Comment:
... whereas this one protects `groot` and a few other globals, again it would be nice if a comment specified all.

What's important here is that the sets of globals controlled by each unparameterized `synchronized` statement are disjoint.